tasks/configure-jenkins.yml

```diff
@@ -2,26 +2,71 @@
 
 - name: Configuration file is up to date (config.xml)
   template:
-    src: "{{ jenkins_source_dir_configs }}/config.xml"
+    src: "{{ jenkins_source_config_xml }}"
     dest: "{{ jenkins_home }}/config.xml"
-    mode: "0770"
+    mode: 0644
+    owner: "{{ jenkins_config_owner }}"
+    group: "{{ jenkins_config_group }}"
 
-- name: Configuration file is up to date (credentials.xml)
+- name: Configure userdata
   template:
-    src: "{{ jenkins_source_dir_configs }}/credentials.xml"
-    dest: "{{ jenkins_home }}/credentials.xml"
-    mode: "0770"
-
-- name: Plugins are installed
-  shell: "curl -X POST \
-    -d '<jenkins><install plugin=\"{{ item }}@latest\" /></jenkins>' \
-    --header 'Content-Type: text/xml' \
-    {{ jenkins_url }}:{{ jenkins_port }}/pluginManager/installNecessaryPlugins"
-  args:
-    creates: "{{ jenkins_home }}/plugins/{{ item }}"
-  with_items: jenkins_plugins
-  register: plugins_result
+    src: "{{ jenkins_source_dir_configs }}/{{ item.src }}"
+    dest: "{{ jenkins_home }}/{{ item.dest }}"
+    mode: 0644
+    owner: "{{ jenkins_config_owner }}"
+    group: "{{ jenkins_config_group }}"
+  when: jenkins_include_userdata
+  with_items: "{{ jenkins_userdata }}"
+
+- name: secrets dir is up to date
+  copy:
+    src: "{{ jenkins_source_secrets }}"
+    dest: "{{ jenkins_home }}/secrets"
+    owner: "{{ jenkins_config_owner }}"
+    group: "{{ jenkins_config_group }}"
+  when: jenkins_include_secrets
```

Comment:
In a similar way, could we do `when: jenkins_source_secrets|default("") != ""` and remove the `jenkins_include_secrets` var?